### ISSUE TITLE
mysql won't shutdown on 10.12

### ISSUE THREAD

**Issue by KingLoui** [Homebrew/homebrew-core#4720]
The installation works just fine and I can use mysql but if I want to shut it down it says shutdown complete but the mysqld process is still active and can't be killed via activity monitor.
I'm not sure if this is the right place to report this issue, I just wanted to raise awareness because I couldn't find anything via google beside this discussion on the apple developer forums.
https://forums.developer.apple.com/thread/50204

Thank you guys for the effort you put into homebrew!

**Comment by zmwangx** [comment]
If you're using the launchd version (`brew services start mysql` or otherwise manually loaded `homebrew.mxcl.mysql.plist`) then launchd will just start it again when you shut it down or kill it. You need to unload it from launchd. Try

```
$ launchctl remove homebrew.mxcl.mysql
```

More info in `man 8 launchd` and `man 1 launchctl`. http://launchd.info/ may be of help, too.

**Comment by KingLoui** [comment]
Unfortunately that didn't help. I start mysql using heroku local (foreman) with the following procfile 
```
web: /usr/local/bin/nginx -c "`pwd`/nginx2.conf" -p "`pwd`"
php: /usr/local/sbin/php-fpm
db:  /usr/local/bin/mysqld
```

when I press ctrl-C all processes stop except for mysqld and foreman won't return to the prompt. The last output from foreman is:
`13:34:33 db.1    |  2016-09-11T11:34:33.111167Z 0 [Note] /usr/local/bin/mysqld: Shutdown complete`

I also tried `brew services stop mysql` to no avail

**Comment by zmwangx** [comment]
Not a foreman user so sorry, I'm not sure.

In any case, I highly doubt this is a Homebrew-specific issue. You can make sure by installing MySQL from another channel, e.g., a binary distribution from http://dev.mysql.com/downloads/ and see if you can reproduce. If not please report back, but otherwise I'd recommend taking it to the right place for support, e.g., #theforeman on Freenode, foreman-users mailing list, MySQL Forums, StackOverflow and such.

Closing this for now, please let us know if you find this to be Homebrew-specific or otherwise have anything that we can help with.

**Comment by KingLoui** [comment]
I just installed mysql 5.7.15 from mysql.com and the shutdown works just fine.
Note that [foreman from ddollar](https://github.com/ddollar/forego) is not theforeman.
Foreman is just a tool to start multiple services at once.

I found out that the problem occurs without the use of foreman. When I use `mysql.server start` and then do `mysql.server stop` the result is still the same that there is a mysqld in my processes although I got the output
```
| => mysql.server stop
Shutting down MySQL
.. SUCCESS! 
```

I checked all the launch agents and demons and did not find any entries for mysql.

**Comment by zmwangx** [comment]
Not sure, but reopened the issue.

**Comment by zmwangx** [comment]
Also it may be something that has been fixed in 5.7.15. https://github.com/Homebrew/homebrew-core/pull/4722

**Comment by KingLoui** [comment]
I would love to try that, can you tell me how I can use that formula when it isn't merged yet?

**Comment by zmwangx** [comment]
Try

```
$ brew update
$ brew pull 4722
$ HOMEBREW_NO_AUTO_UPDATE=1 brew upgrade mysql
```

Don't get your hopes high though.

**Comment by KingLoui** [comment]
I used `brew install https://raw.githubusercontent.com/joeploijens/homebrew-core/c940d5d75dc4fe6a3d9abda6a01f74e40f69977a/Formula/mysql.rb`
works like a charm now 👍

**Comment by zmwangx** [comment]
Interesting, so it is fixed by 5.7.15. Thanks for the confirmation.

Closing this then, please wait until we pull in 5.7.15.

**Comment by zmwangx** [comment]
We have shipped MySQL 5.7.15. `brew update`, reinstall and you should be golden.
